tst/testinstall/MatrixObj/testmatobj_new.g

```diff
@@ -0,0 +1,422 @@
+    local ex, mat, inv;
+
+    ex := ShallowCopy(ex_in);
+    mat := ex.MatObj;
+    inv := mat^(-1);
+    if IsIdentityMat(mat*inv) and IsIdentityMat(inv*mat) then 
+        return true;
+    fi;
+
+    return false;
+end;
+
+# This function requires a filter defining a MatrixObject and a record of 
+# options. 
+# This function runs tests to check whether the specified MatrixObject
+# implementation implements required methods and if these and other methods are 
+# correct (in the sense that they give correct results for the cases tested 
+# here).
+# The following options are supported:
+# randomSource:
+#   See TestMatrixObj_GenerateExample
+# state:
+#   See TestMatrixObj_GenerateExample
+# dimensions: 
+#   See TestMatrixObj_GenerateExample
+# dimensions_red: A list of tuples defining dimensions for matrices. If not set
+#   this is set to [[1,1],[2,2],[3,3],[5,5]]. These dimensions are used to
+#   generate examples for more expensive tests. 
+# domains:
+#   See TestMatrixObj_GenerateExample
+# domains_red:
+#   A list of domains in which matrices should be generated. By default
+#   this is set to [Integers, Rationals, GF(2), GF(5)]. These domains are used 
+#   to generate examples for more expensive tests.
+# reduced_cases: 
+#   A list of triples defining the number of rows, the number of columns and the
+#   domain over which examples are generated. By default this option is unbound.
+#   If bound the options dimensions_red and domains_red are ignored and instead
+#   the cases specified in reduced_cases are used to generate examples for
+#   expensive tests. The reason to use this is that otherwise when specifying
+#   domains and dimensions via domains_red and dimensions_red all combinations 
+#   of the specified dimensions and domains are used. If tests are very 
+#   expensive or just certain combinations should be tested one can use 
+#   reduced_cases instead.
+# customGenerator:
+#   See TestMatrixObj_GenerateExample
+# customInvertibleGenerator:
+#   Like customGenerator a function that generates invertible matrices
+#   compatible with the specified matrix object. It is used when invertible
+#   matrices are required. Default: unbound
+# sourceOfTruthFilter:
+#   See TestMatrixObj_GenerateExample
+# forbidInv:
+#   Boolean set to false by default. If set to true no tests which require
+#   invertible matrices are performed. Intended to either skip potentially
+#   expensive tests or for implementations not able to store invertible 
+#   matrices.
+# breakOnError:
+#   Boolean set to false by default. If set to true Errors or Wrong results
+#   cause a break loop. Otherwise problems are returned as a list of messages.
+# isImmutable:
+#    Boolean set to false by default. If set to true tests of functions which 
+#    change an object and therefore require it to be mutable are skipped. This
+#    option is intended for objects which are always immutable. 
+
+TestMatrixObj := function(filter, optIn)
```

Comment:
What calls `TestMatrixObj` and where?

Reply:
That is the function one should call to execute the tests.